core/lib/state/src/rocksdb/mod.rs

```diff
@@ -60,17 +64,34 @@ impl NamedColumnFamily for StateKeeperColumnFamily {
 pub struct RocksdbStorage {
     db: RocksDB<StateKeeperColumnFamily>,
     pending_patch: InMemoryStorage,
+    enum_index_migration_chunk_size: usize,
 }
 
 impl RocksdbStorage {
     const BLOCK_NUMBER_KEY: &'static [u8] = b"block_number";
+    const ENUM_INDEX_MIGRATION_PROGRESS_KEY: &'static [u8] = b"enum_index_migration_progress";
+    const ENUM_INDEX_MIGRATION_FINISHED: &'static [u8] = b"enum_index_migration_finished";
 
     /// Creates a new storage with the provided RocksDB `path`.
     pub fn new(path: &Path) -> Self {
         let db = RocksDB::new(path, true);
         Self {
             db,
             pending_patch: InMemoryStorage::default(),
+            enum_index_migration_chunk_size: 0,
+        }
+    }
+
+    /// Creates a new storage with the provided RocksDB `path` and enum index migration settings.
+    pub fn new_with_enum_index_migration_chunk_size(
```

Comment:
Why not have a separate method?

```rust
pub fn enable_enum_index_migration(&mut self, chunk_size: usize) {
    self.enum_index_migration_chunk_size = chunk_size;
}
```

Do we want to make this API intentionally awkward to use? 🙃

Reply:
With the new method, you can *replace* this one with

```rust
let mut storage = RocksdbStorage::new(path);
storage.enable_enum_index_migration(chunk_size);
// Do something with `storage`
```

It's more composable this way, and the methods adhere to the single responsibility principle. Sorry for not being clear.

Reply:
Nah, it was clear, I just forgot to remove this one